cpu/native/cli_eui_provider/eui_provider.c

```diff
@@ -0,0 +1,64 @@
+/**
+ * Native CPU EUI provider
+ *
+ * Copyright (C) 2020 Benjamin Valentin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ * @ingroup cpu_native
+ * @{
+ * @file
+ * @author  Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ * @}
+ */
+
+#include <assert.h>
+#include <stdlib.h>
+
+#include "net/l2util.h"
+#include "native_cli_eui_provider.h"
+
+#include "native_internal.h"
+
+/* list of user supplied EUI-64s */
+struct _native_eui64_list;
+static struct _native_eui64_list {
+    eui64_t addr;
+    struct _native_eui64_list *prev;
+} *_eui64_head;
```

Comment:
Why not use `core/clist.h` for this? You can still use it "in reverse" using `clist_rpush()` and the `clist_lpoprpush()` to cycle through the addresses.

Reply:
I'm not convinced this is an improvement / necessary, but I switched to using `list.h` in 73c9b745d0b1851545e4cf032547ebdc43c867bb

Reply:
Experience teaches, re-using code is always an improvement ;-)'